fenix/app/src/androidTest/java/org/mozilla/fenix/ui/ModifierTest.kt

```diff
@@ -0,0 +1,129 @@
+package org.mozilla.fenix.ui
+
+import android.util.Log
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performScrollToIndex
+import androidx.compose.ui.unit.dp
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.compose.ext.onShown
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val ON_SHOWN_ROOT_TAG = "onShownRoot"
+private const val ON_SHOWN_SETTLE_TIME = 1000
+private const val ON_SHOWN_INDEX = 15
+private const val ON_SHOWN_NODE_COUNT = 30
+
+class ModifierTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun verifyModifierOnShownWhenScrolledToWithNoSettleTime() {
+        var onShown = false
+        composeTestRule.setContent {
+            ModifierOnShownContent(
+                settleTime = 0,
+                onVisible = {
+                    onShown = true
+                },
+            )
+        }
+
+        composeTestRule.scrollToOnShownIndex()
+
+        assertTrue(onShown)
+    }
+
+    @Test
+    fun verifyModifierOnShownAfterSettled() {
+        var onShown = false
+        composeTestRule.setContent {
+            ModifierOnShownContent(
+                onVisible = {
+                    Log.d("bondbond", "onvisible")
+                    onShown = true
+                },
+            )
+        }
+
+        composeTestRule.scrollToOnShownIndex()
+
+        assertFalse(onShown)
+
+        composeTestRule.mainClock.advanceTimeUntil(ON_SHOWN_SETTLE_TIME + 5000L) {onShown}
+
+//        composeTestRule.waitUntil(ON_SHOWN_SETTLE_TIME + 5000L) { onShown }
```

Comment:
This is the test I'm currently struggling to find a good way to implement. There isn't a good way to do a true "wait" (nor is it advised), so I'm currently stuck trying to find a good way to have a test cover the settle time of `Modifier.onShown`

Reply:
You can use java.time.Clock interface that's a param for the modifier. The production code uses `SystemClock` and the test can use `FixedClock` so we can control the time for tests.

Reply:
I shall check this out!

Reply:
To close this loop, I was able to get the test to pass once I removed the toolbar logic from the Modifier!